File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
public class Solution {
    public double findMaxAverage(int[] nums, int k) {
        // Calculate the sum of the first window of size k
        double sum = 0;
        for (int i = 0; i < k; i++) {
            sum += nums[i];
        }

        double maxSum = sum;

        // Slide the window over the array
        for (int i = k; i < nums.length; i++) {
            sum = sum - nums[i - k] + nums[i];
            maxSum = Math.max(maxSum, sum);
        }

        // Return the maximum average
        return maxSum / k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.c
double findMaxAverage(int* nums, int numsSize, int k) {
    int sum = 0;
    
    // Initial window sum
    for (int i = 0; i < k; i++) {
        sum += nums[i];
    }
    
    int maxSum = sum;

    // Slide the window
    for (int i = k; i < numsSize; i++) {
        sum += nums[i] - nums[i - k];
        if (sum > maxSum) {
            maxSum = sum;
        }
    }

    return (double)maxSum / k;
}
